spec/resources/api/v2/barcode_printer_resource_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './app/resources/api/v2/barcode_printer_resource'
+
+RSpec.describe Api::V2::BarcodePrinterResource, type: :resource do
+  subject(:resource) { described_class.new(resource_model, {}) }
+
+  let(:resource_model) { build_stubbed :barcode_printer }
+
+  # Test attributes
+  it 'has the expected attributes', :aggregate_failures do
+    expect(resource).not_to have_attribute :id
+    expect(resource).to have_attribute :uuid
+    expect(resource).to have_attribute :name
+    expect(resource).to have_attribute :print_service
+    expect(resource).to have_attribute :barcode_type
+  end
+
+  # Updatable fields
+  it 'allows updating of read-write fields', :aggregate_failures do
+    expect(resource).to have_updatable_field :name
+  end
+
+  it 'disallows updating of read only fields', :aggregate_failures do
+    expect(resource).not_to have_updatable_field :uuid
```

Comment:
To not have updatable field `id` too maybe?

Reply:
I stopped adding `:id` to these simply because I could also check that `:banana` is not an `updatable_field`.  `id` doesn't form part of the resource and so testing for its lack of existence didn't feel like it was adding value to these tests.  These tests are quite poor anyway.  They only test the properties of the `attribute` entries like `readonly`.  They don't confirm whether the resource is immutable and they don't know how to parse the things like `fetchable_fields` and `creatable_fields` methods that are part of the library we're using.  The best tests are in the `requests` spec folder.